src/main/java/picocli/AutoComplete.java

```diff
@@ -793,7 +828,8 @@ private static String generateOptionsCases(List<OptionSpec> argOptionFields, Str
             }
             if (option.completionCandidates() != null) {
                 buff.append(format("%s    %s)\n", indent, concat("|", option.names()))); // "    -u|--timeUnit)\n"
-                buff.append(format("%s      COMPREPLY=( $( compgen -W \"${%s_option_args}\" -- \"%s\" ) )\n", indent, bashify(option.paramLabel()), currWord));
+                buff.append(format("%s      local IFS=$'\\n'\n", indent));
+                buff.append(format("%s      COMPREPLY=( $( compReplyArray \"${%s_option_args[@]}\" ) )\n", indent, bashify(option.paramLabel()), currWord));
```

Comment:
We recently [discovered](https://github.com/remkop/picocli/issues/2053) that the format string has only 2 `%s` variables, but we specify 3 parameters (`indent`, `bashify(option.paramLabel())`, `currWord`)...

This means that the last parameter (`currWord`) is not used.
Is that correct? Can the `currWord` parameter be removed?

Reply:
fixed in #2175